crates/iota-framework/packages/iota-system/tests/rewards_distribution_tests.move

```diff
@@ -972,11 +974,137 @@ module iota_system::rewards_distribution_tests {
         scenario_val.end();
     }
 
+    #[test]
+    fun test_constant_exchange_rates_with_no_rewards() {
+        set_up_iota_system_state();
+        let mut scenario_val = test_scenario::begin(VALIDATOR_ADDR_1);
+        let scenario = &mut scenario_val;
+
+        // Add different stake to different pools
+        stake_with(STAKER_ADDR_1, VALIDATOR_ADDR_1, 1, scenario);
+        stake_with(STAKER_ADDR_2, VALIDATOR_ADDR_2, 50, scenario);
+        stake_with(STAKER_ADDR_3, VALIDATOR_ADDR_3, 3, scenario);
+
+        // need to advance epoch so validator's staking starts counting
+        advance_epoch(scenario);
+
+        // Advance a couple of epochs to make the exchange rate different than 1.
+        // Validator 1: 10% commission.
+        set_commission_rate_and_advance_epoch(VALIDATOR_ADDR_1, 1000, scenario);
+        advance_epoch_with_reward_amounts(0, 1, scenario);
+
+        // Stake and unstake different amounts to one of the pools
+        stake_with(STAKER_ADDR_4, VALIDATOR_ADDR_3, 150, scenario);
+        unstake(STAKER_ADDR_3, 0, scenario);
+
+        // Advance another epoch without rewards
+        advance_epoch_with_reward_amounts(0, 1, scenario);
+        advance_epoch(scenario);
+
+        // Get exchange rates for all pools and check that their rates are constant during the last epoch change
+        scenario.next_tx(VALIDATOR_ADDR_1);
+        let mut system_state = scenario.take_shared<IotaSystemState>();
+
+        let staked_iota_1 = scenario.take_from_address<StakedIota>(STAKER_ADDR_1);
+        let pool_id_1 = staked_iota_1.pool_id();
+        let rates1 = system_state.pool_exchange_rates(&pool_id_1);
+        let pool_token_amount_before_1  = rates1[4].pool_token_amount() as u128;
+        let pool_token_amount_after_1 = rates1[5].pool_token_amount() as u128;
+        let iota_amount_before_1 = rates1[4].iota_amount() as u128;
+        let iota_amount_after_1 = rates1[5].iota_amount() as u128;
```

Comment:
It's not clear to me what the indexes used on rates1 are. Is this the epoch number? If so, you should comment at each point in the code what number epoch we are on, to make it easy to link the points in time we are checking.